libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1444,8 +1444,10 @@ void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const&
 		auto const* tupleType = dynamic_cast<TupleType const*>(&_type);
 		auto const& types = tupleType ? tupleType->components() : vector<TypePointer> { _type.shared_from_this() };
 
-		solAssert(tupleExpression->components().size() == types.size(), "");
-		for (size_t i = 0; i < types.size(); i++)
+		solAssert(tupleExpression->components().size() == types.size() || m_errorReporter.hasErrors(),
+			"The array sizes must match or an error must have been already generated.");
```

Comment:
Perhaps turn it to ``Array sizes don't match or no errors generated.``?

Reply:
Also style: linebreak directly after `(`, every component on a line of its own and `);` on a line of its own.

Reply:
I preferred my original string message, but I am fine with the latter, too. Style fixed.